samtranslator/validator/validator.py

```diff
@@ -7,49 +9,254 @@
+
+            if error_content not in errors_set:
+                # We set the value to None as we don't use it
+                errors_set[error_content] = None
+            return
+
+        for context_error in error.context:
+            # Each context item is also a validation error
+            self._process_error(context_error, errors_set)
+
+    def _cleanup_error_message(self, error):
+        """
+        Cleans an error message up to remove unecessary clutter or replace
+        it with a more meaningful one
+
+        Parameters
+        ----------
+        error : Error
+            Error message to clean
+
+        Returns
+        -------
+        str
+            Cleaned message
+        """
+        final_message = re.sub(self.UNICODE_TYPE_REGEX, r"\1", error.message)
+
+        if final_message.endswith(" under any of the given schemas"):
+            return "Is not valid"
+        if final_message.startswith("None is not of type ") or final_message.startswith("None is not one of "):
+            return "Must not be empty"
+        if " does not match " in final_message and "patternError" in error.schema:
+            return re.sub("does not match .+", error.schema.get("patternError"), final_message)
+
+        return final_message
+
+    def _read_json(self, filepath):
         """
-        Helper method to read a JSON file
-        :param filepath: Path to the file
-        :return dict: Dictionary containing file data
+        Returns the content of a JSON file
+
+        Parameters
+        ----------
+        filepath : str
+            File path
+
+        Returns
+        -------
+        dict
+            Dictionary representing the JSON content
         """
-        with open(filepath, "r") as fp:
+        with open(filepath) as fp:
             return json.load(fp)
+
+
+# Type definition redefinitions
+INTRINSIC_ATTR = {
+    "Fn::And",
+    "Fn::Base64",
+    "Fn::Cidr",
+    "Fn::Equals",
+    "Fn::FindInMap",
+    "Fn::GetAtt",
+    "Fn::GetAZs",
+    "Fn::If",
+    "Fn::ImportValue",
+    "Fn::Join",
+    "Fn::Not",
+    "Fn::Or",
+    "Fn::Select",
+    "Fn::Split",
+    "Fn::Sub",
+    "Fn::Transform",
+    "Ref",
+}
+
+
+def is_object(checker, instance):
+    """
+    'object' type definition
+    Overloaded to exclude intrinsic functions
+
+    Parameters
+    ----------
+    checker : dict
+        Checker
+    instance : element
+        Template element
+
+    Returns
+    -------
+    boolean
+        True if an object, False otherwise
+    """
+    return isinstance(instance, dict) and not has_intrinsic_attr(instance)
+
+
+def is_intrinsic(checker, instance):
+    """
+    'intrinsic' type definition
+
+    Parameters
+    ----------
+    checker : dict
+        [description]
+    instance : [type]
+        [description]
+
+    Returns
+    -------
+    [type]
+        [description]
+    """
+    return isinstance(instance, dict) and has_intrinsic_attr(instance)
+
+
+def has_intrinsic_attr(dict):
```

Comment:
PyLint is probably going to complain overriding built-in dict. Even if it does not, another name might be better.